spring-kafka/src/main/java/org/springframework/kafka/core/DefaultKafkaConsumerFactory.java

```diff
@@ -95,13 +97,27 @@ public void setValueDeserializer(Deserializer<V> valueDeserializer) {
 		return createKafkaConsumer(groupId, clientIdSuffix);
 	}
 
+	@Override
+	public Consumer<K, V> createConsumer(String groupId, String clientIdPrefix, String clientIdSuffix) {
+		return createKafkaConsumer(groupId, clientIdPrefix, clientIdSuffix);
+	}
+
 	protected KafkaConsumer<K, V> createKafkaConsumer() {
 		return createKafkaConsumer(this.configs);
 	}
 
 	protected KafkaConsumer<K, V> createKafkaConsumer(String groupId, String clientIdSuffix) {
-		boolean shouldModifyClientId = this.configs.containsKey(ConsumerConfig.CLIENT_ID_CONFIG)
-				&& clientIdSuffix != null;
+		return createKafkaConsumer(groupId, null, clientIdSuffix);
+	}
+
+	protected KafkaConsumer<K, V> createKafkaConsumer(String groupId, String clientIdPrefix,
+			String clientIdSuffix) {
+		boolean overrideClientIdPrefix = StringUtils.hasText(clientIdPrefix);
+		if (clientIdSuffix == null) {
+			clientIdSuffix = "";
+		}
+		boolean shouldModifyClientId = (this.configs.containsKey(ConsumerConfig.CLIENT_ID_CONFIG)
+				&& clientIdSuffix != null) || overrideClientIdPrefix;
```

Comment:
The `clientIdSuffix` is not null here any more. See `clientIdSuffix = ""` code a line before.

Reply:
Good catch